src/extension-runners/chromium.js

```diff
@@ -0,0 +1,266 @@
+      chromePath: chromiumBinary,
+      chromeFlags: [`--load-extension=${extensions}`],
+      startingUrl,
+    });
+
+    this.chromiumInstance.process.once('close', () => {
+      this.chromiumInstance = null;
+
+      if (!this.exiting) {
+        log.info('Exiting on Chromium instance disconnected.');
+        this.exit();
+      }
+    });
+  }
+
+  wssBroadcast(data: Object) {
+    for (const client of this.wss.clients) {
+      if (client.readyState === WebSocket.OPEN) {
+        client.send(JSON.stringify(data));
+      }
+    }
+  }
+
+  async createReloadManagerExtension() {
+    const tmpDir = new TempDir();
+    await tmpDir.create();
+    this.registerCleanup(() => tmpDir.remove());
+
+    const extPath = path.join(
+      tmpDir.path(),
+      `reload-manager-extension-${Date.now()}`
+    );
+
+    log.debug(`Creating reload-manager-extension in ${extPath}`);
+
+    await asyncMkdirp(extPath);
+
+    await fs.writeFile(
+      path.join(extPath, 'manifest.json'),
+      JSON.stringify({
+        manifest_version: 2,
+        name: 'web-ext Reload Manager Extension',
+        version: '1.0',
+        permissions: ['management', 'tabs'],
+        background: {
+          scripts: ['bg.js'],
+        },
+      })
+    );
+
+    const bgPage = `function bgPage(webExtParams) {
+      async function getAllDevAddons() {
+        const [managerAddon, allAddons] = await Promise.all([
+          new Promise(r => chrome.management.getSelf(r)),
+          new Promise(r => chrome.management.getAll(r)),
+        ]);
+
+        return allAddons.filter((addon) => {
+          return addon.installType === "development" &&
+            addon.id !== managerAddon.id;
+        });
+      }
+
+      const setEnabled = (addonId, value) =>
+        new Promise(r => chrome.management.setEnabled(addonId, value, r));
+
+      async function reloadAddon(addonId) {
+        await setEnabled(addonId, false);
+        await setEnabled(addonId, true);
+      }
+
+      const ws = new window.WebSocket([
+        "ws://", webExtParams.address, ":", webExtParams.port
+      ].join(""));
+
+      ws.onmessage = async (evt) => {
+        const msg = JSON.parse(evt.data);
+        if (msg.type === 'webExtReloadAllExtensions') {
+          const {extension} = msg;
+
+          const devExtensions = await getAllDevAddons();
+          for (var ext of devExtensions) {
+            reloadAddon(ext.id);
+          }
+        }
+      };
+    }`;
+
+    await fs.writeFile(path.join(extPath, 'bg.js'),
+                       `(${bgPage})(${JSON.stringify(this.wss.address())})`
+    );
+    return extPath;
+  }
+
+  /**
+   * Reloads all the extensions, collect any reload error and resolves to
+   * an array composed by a single ExtensionRunnerReloadResult object.
+   */
+  async reloadAllExtensions(): Promise<Array<ExtensionRunnerReloadResult>> {
+    const runnerName = this.getName();
+
+    this.wssBroadcast({
+      type: 'webExtReloadAllExtensions',
+    });
+
```

Comment:
Shouldn't we wait until the reload has been processed by the browser before resolving the promise?

Reply:
Ideally yes, but I'm currently leaving it as one of the improvement we may introduce in a followup.

Reply:
also filed as #1686